### DISCUSSION TITLE
Run scripts within scripts in a separate kenv

### DISCUSSION THREAD

**Discussion by zacjones93** [johnlindquist/kit#350] [Q&A] [answered]
I have a separate kenv `egghead` that I am building scripts out in. 

I was thinking about how I could recursively run a script so that the user can keep picking options until they want to exit out of the script on their own. 

I ran into an issue where kit only looks in my home kenv for scripts that are invoked with `run`. Is there a way I can invoke a script within a separate kenv?

Intuitively, I tried to prepend the kenv to the script:

```js
await run("egghead/run-me");
```

Happy to try another approach to looping scripts but this was the first way I thought of.

**Comment by johnlindquist** [comment, marked as answer]
Ah, good catch. I like your intuition approach. 👍 I'll put that in a build tonight

**Reply by johnlindquist** [reply]
You can try passing in the full path today though: (`kenvPath` is a helper to resolve paths in your kenv)

```js
await run(
  kenvPath(
    "kenvs",
    "egghead",
    "scripts",
    "run-me.js"
  )
)

```

**Reply by zacjones93** [reply]
ahhhh that makes sense! Thanks
